storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -0,0 +1,604 @@
+            @Override
+            public boolean test(String obj) {
+                return obj != null ? getTopologyNameById(obj).equals(topologyName) : false;
+            }
+        };
+        return Utils.findOne(pred, activeTopologys);
+    }
+
+    public static Map<String, StormBase> topologyBases(IStormClusterState stormClusterState) {
+        return _instance.topologyBasesImpl(stormClusterState);
+    }
+
+    protected Map<String, StormBase> topologyBasesImpl(IStormClusterState stormClusterState) {
+        List<String> activeTopologys = stormClusterState.activeStorms();
+        Map<String, StormBase> stormBases = new HashMap<String, StormBase>();
+        if (activeTopologys != null) {
+            for (String topologyId : activeTopologys) {
+                StormBase base = stormClusterState.stormBase(topologyId, null);
+                if (base != null) {
+                    stormBases.put(topologyId, base);
+                }
+            }
+        }
+        return stormBases;
+    }
+
+    public static void validateDistributedMode(Map conf) {
+        if (ConfigUtils.isLocalMode(conf)) {
+            throw new IllegalArgumentException("Cannot start server in local mode!");
+        }
+    }
+
+    private static void validateIds(StormTopology topology) throws InvalidTopologyException {
+        List<String> componentIds = new ArrayList<String>();
+
+        for (StormTopology._Fields field : Thrift.getTopologyFields()) {
+            if (ThriftTopologyUtils.isWorkerHook(field) == false) {
+                Object value = topology.getFieldValue(field);
+                if (value != null) {
+                    Map<String, Object> componentMap = (Map<String, Object>) value;
+                    componentIds.addAll(componentMap.keySet());
+
+                    for (String id : componentMap.keySet()) {
+                        if (Utils.isSystemId(id)) {
+                            throw new InvalidTopologyException(id + " is not a valid component id.");
+                        }
+                    }
+                    for (Object componentObj : componentMap.values()) {
+                        ComponentCommon common = getComponentCommon(componentObj);
+                        Set<String> streamIds = common.get_streams().keySet();
+                        for (String id : streamIds) {
+                            if (Utils.isSystemId(id)) {
+                                throw new InvalidTopologyException(id + " is not a valid stream id.");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        List<String> offending = Utils.getRepeat(componentIds);
+        if (offending.isEmpty() == false) {
+            throw new InvalidTopologyException("Duplicate component ids: " + offending);
+        }
+    }
+
+    private static boolean isEmptyInputs(ComponentCommon common) {
+        if (common == null) {
+            return true;
```

Comment:
The original code would throw an NPE if this was null.  In this case I can see it being a minor nit, but I still would prefer to keep the code the same if possible.